### ISSUE TITLE
Support for String enums

### ISSUE THREAD

**Issue by b-eyselein** [scala-tsi/scala-tsi#237]
At the moment this library supports enums only as numeric enums, e. g.
```typescript
export enum MyEnum {
  X,
  Y
}
```

But Typescript also support enums as [string enums](https://www.typescriptlang.org/docs/handbook/enums.html#string-enums) in this form:
```typescript
export enum MyStringEnum {
  X = 'X',
  Y = 'Y'
}
```

Especially when serializing enums to JSON the latter form is more common (e. g. with the Play Framework Scala Json Library). Can you please add support for string enums? You could preserve the current functionality with a new class `TSStringEnum` besides the current `TSEnum`.

If you want, I can prepare a merge request.

**Comment by dhoepelman** [comment]
Thanks for the suggestion @b-eyselein , a merge request to support the syntax would definetely be accepted.

Note that scala/java enums are already supported as a union of strings, [which is a little more common in my experience](https://blog.bam.tech/developer-news/should-you-use-enums-or-union-types-in-typescript).

[Scala example](https://github.com/scala-tsi/scala-tsi/blob/master/example/src/main/scala/models/enumeration/ScalaEnum.scala), [Typescript output](https://github.com/scala-tsi/scala-tsi/blob/master/example/expected_model.ts#L57).

**Comment by dhoepelman** [comment]
Released in 0.7.0
